### PR TITLE
[videoplayer] Show stream flags in the gui and some fixes/improvements.

### PR DESCRIPTION
**Pull request by ace20022**
## TODO:
- [x] Define and use new string ids
- [x] more test, e.g., bds

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -496,7 +497,7 @@ void CSelectionStreams::Update(CDVDInputStream* input, CDVDDemux* demuxer, std::
 
       DVDNavSubtitleStreamInfo info = nav->GetSubtitleStreamInfo(i);
       s.name     = info.name;
-      s.flags = info.flags;
+      s.flags    = info.flags;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: no vertical alignment please

File: xbmc/cores/IPlayer.h [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "VideoSettings.h"
 #include "guilib/Geometry.h"
 #include "guilib/Resolution.h"
+#include "VideoPlayer/DVDDemuxers/DVDDemux.h"
```
Comment (FernetMenta) [inline xbmc/cores/IPlayer.h]: any definition from nested folders is not supposed to be exposed to the world. there is a new folder "Interface". please move/create required definitions to this folder.

Comment (ace20022) [reply xbmc/cores/IPlayer.h]: I didn't like it and wanted to ask you anyway, thx for the hint.

File: xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp [hunk 3]
```diff
@@ -524,3 +526,30 @@ std::string CGUIDialogAudioSubtitleSettings::SettingFormatterPercentAsDecibel(st
...
+    localizedFlags.emplace_back(g_localizeStrings.Get(37000));
+
+  for (auto &flag : localizedFlags)
+  {
+    formated += flag;
+    formated += ", ";
+  }
+  if (!formated.empty())
+  {
+    formated = formated.substr(0, formated.length() - 2);
```
Comment (tamland) [inline xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp]: StringUtils::Join

Comment (ace20022) [reply xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp]: Nice, thx!

**Comment by ace20022** [comment]
@FernetMenta like that, structurewise?

**Comment by FernetMenta** [comment]
@ace20022 much better than what we had before but we need to keep in mind that there is work ahead. Currenty IPlayer is used by PAPlayer and RetroPlayer too but there are many methods only implemented by VideoPlayer. IPlayer depends on a file owned by VP, which is not nice but ok for now.
I think in the future we should reduce the IPlayer interface to a common set of methods shared by all players. If some code wants to access VP specific methods, it can query IPlayer for a specific interface.

**Comment by ace20022** [comment]
@FernetMenta could you please check it? I think I've added everything I wanted. @MartijnKaijser are the strings.po changes ok?

**Review by FernetMenta — APPROVED** [review]

**Comment by MartijnKaijser** [comment]
Changes look fine if they're indeed not used anywhere else.

**Pull request merged**